### DISCUSSION TITLE
time.Sleep

### DISCUSSION THREAD

**Discussion by hghgsat** [brutella/hkknx-public#127] [Q&A] [answered]
Hi,

time.Sleep(1) did not work for me (undefined symbol 'Sleep')

Here is my simple script:

```go
var hkknx, time = import("hkknx"), import("time")

var i = 1

for i==1 {

	// windspeed
	var buf, err = hkknx.GroupRead("1/1/2")
	if err != nil {
		println(err)
	} else {
		println(buf)
		//println(hkknx.ParseDPT9005(buf)) 
	}

	// sleep
	time.Sleep(1)

	//i++
}
```
and for reading the windspeed of my Quadra Weather station it would be great to have ParseDPT9005

**Comment by brutella** [comment, marked as answer]
> time.Sleep(1) did not work for me (undefined symbol 'Sleep')

`time.Sleep` was intentionally left out for scripts because the execution of `time.Sleep` cannot be stopped.
For example when you run a script, which executes `time.Sleep(1 * time.Hour)`, you won't be able to stop it and the script will run for one hour. Please use `<-time.After(1 * time.Hour)` instead.

> and for reading the windspeed of my Quadra Weather station it would be great to have ParseDPT9005

I've just created an issue for that. #128 
For now you can use any of the 9.00x methods to parse the data, for example [ParseDPT9001](https://hochgatterer.me/pkg/hkknx/#ParseDPT9001).

**Comment by hghgsat** [comment]
danke - du nutzt das sleep in einem beispiel "Beispiel: Zeitverzögertes Schalten" - sollte dort vlt. auch umgebaut werden

**Comment by brutella** [comment]
Danke für den Hinweis. Hab ich geändert.
